src/platforms/desktop-opengl.c

```diff
@@ -90,6 +90,17 @@ void platform_init(void) {
     if (SDL_Init(SDL_INIT_VIDEO | SDL_INIT_AUDIO) != 0) {
         log_fatal("Error initializing SDL");
     }
+    
+    // Set the scaling algorithm to nearest-neighbor for crisp pixel art rendering.
+    // This ensures that pixel art graphics remain sharp when scaled.
+    SDL_SetHint(SDL_HINT_RENDER_SCALE_QUALITY, "0");

```

Comment:
It's strange you have to set this explicitly because it is the default. What version of SDL are you using?

Reply:
SDL `2.30.8`.

Reply:
Interesting. Let me try updating my Mac to that version.

Reply:
Can you confirm that this hint is the minimum required fix? I'd prefer not to take more changes than needed. Hopefully you understand.

Reply:
You're right; it's working without the first two hints.